code/game/objects/structures/bedsheet_bin.dm

```diff
@@ -29,6 +30,22 @@ LINEN BINS
 		..()
 
 /obj/item/bedsheet/attack_self(mob/user)
+	if(bedsuffix)
+		var/newbedpath = "/obj/item/bedsheet/adjusted/" + bedsuffix
```

Comment:
There's no need for this to not just be a typepath

Reply:
> There's no need for this to not just be a typepath
Show me how great nicholas